### ISSUE TITLE
Support C/F mixed case for least-square problem

### ISSUE THREAD

**Issue by termoshtt** [rust-ndarray/ndarray-linalg#234]
From #227 

```
let a: Array2<f64> = random((3, 4).f());
let b: Array2<f64> = random((3, 2));
let result = a.least_squares(&b).unwrap();
```

will returns `Lapack(InvalidShape)` error, but it should be transposed automatically.

**Comment by termoshtt** [comment]
Fixed in #220, this issue will be closed after #206 is merged

**Comment by termoshtt** [comment]
Done in #220 and merged by #206
